commons/src/main/java/io/aiven/kafka/connect/common/source/task/PartitionPathTaskAssignmentStrategy.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.source.task;
+
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PartitionPathTaskAssignmentStrategy allows source connectors to distribute tasks based off the folder structure that
+ * partition number that is defined in that structure. for example /PREFIX/partition={{partition}}/YYYY/MM/DD/mm this
+ * will split all tasks by the number of unique partitions defined in the storage path. e.g. Task distribution in
+ * Connect with 10 Partitions and 3 tasks |Task | Partition| |0|0| |1|1| |2|2| |0|3| |1|4| |2|5| |0|6| |1|7| |2|8| |0|9|
+ */
+public final class PartitionPathTaskAssignmentStrategy implements TaskAssignmentStrategy {
+    public static final String PARTITION_NUMBER_PATTERN = "\\{\\{partition}}";
```

Comment:
```suggestion
    public static final String PARTITION_ID_PATTERN = "\\{\\{partition}}";
```
id seems more close to kafka terms